src/contracts/core/AllocationManager.sol

```diff
@@ -0,0 +1,733 @@
+                    if (block.timestamp >= pendingFreeMagnitude.completableTimestamp) {
+                        break;
+                    }
+
+                    // slash the pending deallocation proportionally and store the updated magnitudeDiff
+                    uint64 slashedAmount =
+                        uint64(uint256(bipsToSlash) * uint256(pendingFreeMagnitude.magnitudeDiff) / BIPS_FACTOR);
+                    pendingFreeMagnitude.magnitudeDiff -= slashedAmount;
+                    // keep track of total slashed magnitude
+                    slashedMagnitude += slashedAmount;
+                }
+            }
+
+            // 3. update totalMagnitude, get total magnitude and subtract slashedMagnitude
+            // this will be reflected in the conversion of delegatedShares to shares in the DM
+            Snapshots.History storage totalMagnitudes = _totalMagnitudeUpdate[operator][strategies[i]];
+            totalMagnitudes.push({key: uint32(block.timestamp), value: totalMagnitudes.latest() - slashedMagnitude});
+        }
+    }
+
+    /**
+     * @notice Called by an operator to cancel a salt that has been used to register with an AVS.
+     *
+     * @param salt A unique and single use value associated with the approver signature.
+     */
+    function cancelSalt(
+        bytes32 salt
+    ) external override {
+        // Mutate `operatorSaltIsSpent` to `true` to prevent future spending.
+        operatorSaltIsSpent[msg.sender][salt] = true;
+    }
+
+    /**
+     * @notice Helper for setting an operators allocation delay.
+     * @param operator The operator to set the delay on behalf of.
+     * @param delay The allocation delay in seconds.
+     */
+    function _setAllocationDelay(address operator, uint32 delay) internal {
+        require(delay != 0, InvalidDelay());
+
+        AllocationDelayInfo storage delayInfo = _allocationDelayInfo[operator];
+
+        bool pendingInEffect = delayInfo.pendingDelay != 0 && block.timestamp >= delayInfo.pendingDelayEffectTimestamp;
+
+        if (pendingInEffect) {
+            delayInfo.delay = delayInfo.pendingDelay;
+        }
+
+        delayInfo.pendingDelay = delay;
+        delayInfo.pendingDelayEffectTimestamp = uint32(block.timestamp + ALLOCATION_DELAY_CONFIGURATION_DELAY);
+
+        emit AllocationDelaySet(operator, delay);
+    }
+
+    /**
+     * @notice For a single strategy, update freeMagnitude by adding completable pending free magnitudes
+     * @param operator address to update freeMagnitude for
+     * @param strategy the strategy to update freeMagnitude for
+     * @param numToComplete the number of pending free magnitudes deallocations to complete
+     * @dev read through pending free magnitudes and add to freeMagnitude if completableTimestamp is >= block timestamp
+     * In addition to updating freeMagnitude, updates next starting index to read from for pending free magnitudes after completing
```

Comment:
I found this wording to be really, really hard to parse. here's my attempt to reword:
`In addition to updating freeMagnitude, the index of the next pending free magnitude is also updated` -- idk if it's great, but I think this is an improvement at least

Reply:
Impl and natspec has changed